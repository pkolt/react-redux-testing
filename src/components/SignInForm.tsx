import React from 'react';
import Button from './Button';

export interface SignInFormData {
    login: string;
    password: string;
}

interface SignInFormProps {
    onSubmit: (data: SignInFormData) => void;
}

const SignInForm: React.FC<SignInFormProps> = (props: SignInFormProps) => {
    const {onSubmit} = props;
    const [login, setLogin] = React.useState('');
    const [password, setPassword] = React.useState('');

    const handleSubmit = (event: React.FormEvent) => {
        event.preventDefault();
        onSubmit({login, password});
    };

    const handleInputLogin = (event: React.FormEvent<HTMLInputElement>) => setLogin(event.currentTarget.value);
    const handleInputPassword = (event: React.FormEvent<HTMLInputElement>) => setPassword(event.currentTarget.value);

    return (
        <form onSubmit={handleSubmit}>
            <div className="form-group">
              <label htmlFor="login">Пользователь</label>
              <input
                type="text"
                name="login"
                id="login"
                className="form-control"
                onChange={handleInputLogin}
              />
            </div>
            <div className="form-group">
              <label htmlFor="password">Пароль</label>
              <input
                type="password"
                name="password"
                id="password"
                className="form-control"
                onChange={handleInputPassword}
              />
            </div>
            <div className="d-flex justify-content-center">
              <Button modifier="btn-primary">Войти</Button>
            </div>
          </form>
    );
}

export default SignInForm;
