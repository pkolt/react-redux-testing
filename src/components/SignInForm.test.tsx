import React from 'react';
import SignInForm from './SignInForm';
import { shallow } from 'enzyme';

describe('<SignInForm />', () => {
  it('renders without crashing', () => {
    const mockOnSubmit = jest.fn();
    const wrapper = shallow(<SignInForm onSubmit={mockOnSubmit} />);
    expect(wrapper).toMatchSnapshot();
  });
});
