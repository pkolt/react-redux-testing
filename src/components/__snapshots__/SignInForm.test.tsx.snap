// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInForm /> renders without crashing 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="login"
    >
      Пользователь
    </label>
    <input
      className="form-control"
      id="login"
      name="login"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="password"
    >
      Пароль
    </label>
    <input
      className="form-control"
      id="password"
      name="password"
      onChange={[Function]}
      type="password"
    />
  </div>
  <div
    className="d-flex justify-content-center"
  >
    <Button
      modifier="btn-primary"
    >
      Войти
    </Button>
  </div>
</form>
`;
